library/manageiq_alert.py

```diff
@@ -0,0 +1,282 @@
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  miq_verify_ssl:
+    description:
+      - whether SSL certificates should be verified for HTTPS requests
+    required: false
+    default: True
+    choices: ['True', 'False']
+  ca_bundle_path:
+    description:
+      - the path to a CA_BUNDLE file or directory with certificates
+    required: false
+    default: null
+  description:
+    description:
+      - the alert definition description in manageiq
+    required: false
+    default: null
+  entity:
+    description:
+      - the entity to base the alert on in manageiq
+    required: false
+    choices: ['node', 'vm', 'host', 'storage', 'cluster', 'ems', 'miq_server', 'middleware_server']
+    default: null
+  expression:
+    description:
+      - the expression to evaluate
+    required: false
+    default: null
+  options:
+    description:
+      - Additional alert options, including the notification type and frequency
+    required: false
+    default: null
+  enabled:
+    description:
+      - whether the alert is active or not
+    required: false
+    default: true
+  state:
+    description:
+      - the state of the alert definition
+      - On present, it will create the alert if it does not exist, or update it
+        if the associated data is different
+      - On absent, it will delete the alert if it exists
+    required: false
+    choices: ['present', 'absent']
+    default: 'present'
+'''
+
+EXAMPLES = '''
+# Create a new alert in ManageIQ
+  manageiq_alert:
+    description: Test Alert 01
+    options:
+      notifications:
+        delay_next_evaluation: 60
+        evm_event: {}
+    entity: node
+    expression:
+      eval_method: dwh_generic
+      mode: internal
+    enabled: true
+    state: present
+    miq_url: 'http://localhost:3000'
+    miq_username: 'admin'
+    miq_password: '******'
+    miq_verify_ssl: False
+'''
+
+import os
+from manageiq_client.api import ManageIQClient as MiqApi
+
+
+class ManageIQAlert(object):
+    """ ManageIQ object to execute alert definitions management operations in manageiq
+
+    url            - manageiq environment url
+    user           - the username in manageiq
+    password       - the user password in manageiq
+    miq_verify_ssl - whether SSL certificates should be verified for HTTPS requests
+    ca_bundle_path - the path to a CA_BUNDLE file or directory with certificates
+    """
+
+    supported_entities = {
+        'node': 'ContainerNode', 'vm': 'Vm', 'miq_server': 'MiqServer', 'host': 'Host',
+        'storage': 'Storage', 'cluster': 'EmsCluster', 'ems': 'ExtManagementSystem',
+        'miq_server': 'MiqServer', 'middleware_server': 'MiddlewareServer'
```

Comment:
I would rename this to `entity_to_miq_entity_map` and change all the leftover `db` vars and parameters to `miq_entity`